src/spotify-card.js

```diff
@@ -58,6 +59,37 @@ class PlayerSelect extends Component {
   render() {
     const { devices } = this.props;
     const { chromecastDevices } = this.state;
+    const choice_form = html`<div class="dropdown-content">
+          <a onClick=${() => {}}><i>Spotify Connect devices</i></a>
+          ${devices.map(
+            (device, idx) => html`
+              <a onClick=${() => this.selectDevice(device)} style="margin-left: 15px">${device.name}</a>
+            `
+          )}
+          <a onClick=${() => {}}><i>Chromecast devices</i></a>
+          ${chromecastDevices.map(
+            chromecastDevice => html`
+              <a onClick=${() => this.selectChromecastDevice(chromecastDevice)} style="margin-left: 15px">${chromecastDevice.name + ' (' + chromecastDevice.cast_type + ')'}</a>
+            `
+          )}
+        </div>`;
+
+    let form = '';
+    if (this.props.player && this.props.player == this.state.selectedDevice) {
```

Comment:
I think you can place this in the  `componentWillReceiveProps` method unless you want to have the possibility to show the menu if the player doesn't exist anymore (which would require a little more logic)

Reply:
Mhh maybe, it's my  very first steps with lovelace and I'm not really a JS developer.

Reply:
So, just read a bit about React components. So, if I understand correctly, `componentWillReceiveProps` will be called once and only once each time the props are updated.

If I move the code to `componentWillReceiveProps`, I then need to share `form` so that `render` can access it. Is `this.state` the way to go ? Doesn't sound like a good idea but, as said, I'm not really a JS dev